tests_backend/run_backend_tests.ts

```diff
@@ -99,12 +108,19 @@ async function startSupabaseBackend(
   redis: 'none' | 'local' | 'upstash' = 'none',
   envFile?: string,
 ) {
+  backendType = redis
+
+  if (redis === 'upstash' && !envFile) {
+    p.log.error('Cannot start upstash backend without env file')
+    Deno.exit(1)
+  }
+
   const command = new Deno.Command('supabase', {
     args: [
       'functions',
       'serve',
       '--env-file',
-      (redis !== 'upstash' || !envFile) ? `supabase/.env.exemple${redis === 'local' ? '.redis' : ''}` : envFile,
+      (redis !== 'upstash' || !envFile) ? (redis === 'none' ? noRedisEnvFilePath : localRedisEnvFilePath) : envFile,
```

Comment:
the parentheses that you put in there already made pretty clear what this data is, but instead of two ternary expressions we could have the first one already solved

```suggestion
      (redis !== 'upstash' || !envFile) ? envPathCredtedPreviously : envFile,
```

Reply:
I do not understand what are out asking for to be honest

Reply:
I do not see how this `envPathCredtedPreviously` could improve the codebase, could you please explain it?

Reply:
yeah sure

the thing is that having two ternary expressions can lead to future issues in maintainability and readability

it would make it more accessible to solve the expression `redis === 'none' ? noRedisEnvFilePath : localRedisEnvFilePath` first in a variable and then use its value, rather than `expression1 ? expression2 ? value : otherValue : falseValue`

it's easy to read now cause it's fresh, but it definitely can bring some confusion to less experienced developers